bindings/core/src/method_handler/utils.rs

```diff
@@ -111,18 +111,26 @@ pub(crate) fn call_utils_method_internal(method: UtilsMethod) -> Result<Response
             let output = Output::try_from(output)?;
             Response::HexBytes(prefix_hex::encode(output.pack_to_vec()))
         }
-        UtilsMethod::VerifyTransactionSemantic { inputs, transaction } => {
-            let conflict = verify_semantic(
-                &inputs
-                    .into_iter()
-                    .map(InputSigningData::try_from)
-                    .collect::<iota_sdk::client::Result<Vec<InputSigningData>>>()?,
-                &SignedTransactionPayload::try_from_dto(transaction)?,
-            )?;
-            // TODO conflict reason is an Option now
-            todo!();
-            // Response::TransactionFailureReason(conflict)
+        UtilsMethod::VerifyTransactionSemantic {
+            transaction,
+            inputs,
+            unlocks,
+        } => {
+            let transaction = Transaction::try_from_dto(transaction)?;
+            let inputs = &inputs
+                .into_iter()
+                .map(InputSigningData::try_from)
+                .collect::<iota_sdk::client::Result<Vec<InputSigningData>>>()?;
+            let inputs = inputs
+                .iter()
+                .map(|input| (input.output_id(), &input.output))
+                .collect::<Vec<(&OutputId, &Output)>>();
```

Comment:
Why iterate twice?

Reply:
The first one is to collect and return the conversion error, I didn't want to bother much because we're supposedly removing the DTO soon